### PR TITLE
[KL-338] Fix network confirmation page

### PR DESCRIPTION
**Pull request by MuKoJla**

### CODE REVIEW

File: libs/ui/src/components/navigator/components/header-close-button/header-close-button.tsx [hunk 1]
```diff
@@ -24,11 +25,19 @@ const goBackRoutes = [
...
+interface Props {
+  onCloseButtonPress?: OnEventFn<void>;
+}
+
+export const HeaderCloseButton: FC<Props> = ({ onCloseButtonPress }) => {
   const { navigate, goBack } = useNavigation();
   const { name } = useRoute<RouteProp<ScreensParamList>>();
 
   const onClosePress = () => {
+    if (isDefined(onCloseButtonPress)) {
```
Comment (unixvb) [inline libs/ui/src/components/navigator/components/header-close-button/header-close-button.tsx:37]: you could set default value for  `onCloseButtonPress = emptyFn` and then omit this checks

File: libs/ui/src/modals/screens/network-change-confirmation/network-change-confirmation.tsx [hunk 2]
```diff
@@ -31,68 +33,76 @@ const changeNetworkRules: AllowsRules[] = [
 ];
 
 export const NetworkChangeConfirmation: FC = () => {
-  const { navigate } = useNavigation();
-  const { name: selectedNetworkName } = useSelectedNetworkSelector();
+  const selectedNetwork = useSelectedNetworkSelector();
+  const showLoader = useShowLoaderSelector();
+  const { name: selectedNetworkName, iconName: selectedNetworkIcon } = useMemo(() => selectedNetwork, []);
```
Comment (unixvb) [inline libs/ui/src/modals/screens/network-change-confirmation/network-change-confirmation.tsx]: what does this hook do?

Comment (unixvb) [reply libs/ui/src/modals/screens/network-change-confirmation/network-change-confirmation.tsx]: do not do so much destructurization, it's hard to read.  just use `selectedNetwork.name` and `selectedNetwork.iconName`

Comment (MuKoJla) [reply libs/ui/src/modals/screens/network-change-confirmation/network-change-confirmation.tsx]: It remembers the selected network before change

File: libs/ui/src/components/navigator/components/header-close-button/header-close-button.tsx (hunk 1 above)
Comment (MuKoJla) [reply libs/ui/src/components/navigator/components/header-close-button/header-close-button.tsx:37]: But I need to do check, and I only need `onCloseButtonPress` to occur and I have `return` here

File: libs/ui/src/modals/screens/network-change-confirmation/network-change-confirmation.tsx (hunk 2 above)
Comment (MuKoJla) [reply libs/ui/src/modals/screens/network-change-confirmation/network-change-confirmation.tsx]: image

**Review by unixvb — APPROVED** [review]

**Review by Boltman92 — APPROVED** [review]

**Review by yarmoshevych — APPROVED** [review]

**Pull request merged**